svc.go

```diff
@@ -243,15 +234,62 @@ func (as *autoNATService) background(ctx context.Context) {
 	}
 }
 
-func (as *autoNATService) extractPort(m ma.Multiaddr) (name string, port string, err error) {
-	for _, p := range m.Protocols() {
-		if p.Code == ma.P_TCP || p.Code == ma.P_UDP {
-			v, err := m.ValueForProtocol(p.Code)
+//replace obsaddr's port number with the port number of a
+func patchObsaddr(a ma.Multiaddr, obsaddr ma.Multiaddr) (bool, ma.Multiaddr) {
```

Comment:
consider a return signature of `(ma.Multiaddr, error)` with an error of `nil` indicating that the patch was successful?